### ISSUE TITLE
Multiple mapping with prefix breaks

### ISSUE THREAD

**Issue by jackielii** [kovidgoyal/kitty#6877]
**Describe the bug**

Multiple mapping with prefix breaks. Seems only the last one mapped the prefix the the command

**To Reproduce**
```
❯ git rev-parse --short=8 HEAD
f2075f99
kitty on  master [✘?] via  v1.21.2 via  v3.11.6 
❯ ./dev.sh build
Build successful. Run kitty as: kitty/launcher/kitty
kitty on  master [✘?] via  v1.21.2 via  v3.11.6 
❯ ~/personal/kitty/kitty/launcher/kitty -c none -o 'map ctrl+a>v launch --cwd=current --location split' -o 'map ctrl+a>c clear_terminal to_cursor active'```

press ctrl-a does `clear_terminal` instead of as a prefix

**Comment by kovidgoyal** [comment]
Doesnt repro for me on Linux. Running with

kitty -c none -o 'map ctrl+a>v launch --cwd=current --location split' -o 'map ctrl+a>c clear_terminal to_cursor
active' --debug-input

Pressing ctrl+a then releasing all keys then pressing c shows in the
debug output the line

```
on_key_input: glfw key: 0x63 native_code: 0x63 action: PRESS mods: none text: 'c' state: 0
KeyPress matched action: clear_terminal
handled as shortcut
```

as expected. Are you on macos?

**Comment by jackielii** [comment]
Double checked on the latest commit, and adding the --debug-input flag. Still the same. Here is the commands and logs:

```
kitty on  master [✘?] via  v1.21.2 via  v3.11.6 took 7s 
❯ git rev-parse --short=8 HEAD
53980d00
kitty on  master [✘?] via  v1.21.2 via  v3.11.6 
❯ ./dev.sh build
Build successful. Run kitty as: kitty/launcher/kitty
kitty on  master [✘?] via  v1.21.2 via  v3.11.2 
❯ ./kitty/launcher/kitty -c none -o 'map ctrl+a>v launch --cwd=current --location split' -o 'map ctrl+a>c clear_terminal to_cursor active' --debug-input
on_focus_change: window id: 0x1 focused: 1
Move x: 1486.1 y: 2078.8 grabbed: 0
[335 13:04:54.864162] WARNING: Your system's OpenGL implementation does not have glCopyImageSubData, falling back to a slower implementation
-------------- flags changed -----------------
NSEvent: type=FlagsChanged loc=(477.051,324.605) time=440989.3 flags=0x40101 win=0x108724460 winNum=60958 ctxt=0x0 keyCode=59
flagsChanged: modifier: ctrl native_key: 0x3b (<noname>) glfw_key: 0xe062 mods: ctrl 
on_key_input: glfw key: 0xe062 native_code: 0x3b action: PRESS mods: ctrl text: '' state: 0 ignoring as keyboard mode does not support encoding this event
---------------- key down -------------------
NSEvent: type=KeyDown loc=(477.051,324.605) time=440989.8 flags=0x40101 win=0x108724460 winNum=60958 ctxt=0x0 chars="" unmodchars="a" repeat=0 keyCode=0
keyDown triggered global menu bar action ignoring
----------------- key up --------------------
NSEvent: type=KeyUp loc=(477.051,324.605) time=440990.0 flags=0x40101 win=0x108724460 winNum=60958 ctxt=0x0 chars="" unmodchars="a" repeat=0 keyCode=0
keyUp ignored as corresponds to previous keyDown that triggered a shortcut
-------------- flags changed -----------------
NSEvent: type=FlagsChanged loc=(477.051,324.605) time=440990.0 flags=0x100 win=0x108724460 winNum=60958 ctxt=0x0 keyCode=59
flagsChanged: modifier: ctrl native_key: 0x3b (<noname>) glfw_key: 0xe062 mods: none 
on_key_input: glfw key: 0xe062 native_code: 0x3b action: RELEASE mods: none text: '' state: 0 ignoring as keyboard mode does not support encoding this event
---------------- key down -------------------
NSEvent: type=KeyDown loc=(477.051,324.605) time=440990.8 flags=0x100 win=0x108724460 winNum=60958 ctxt=0x0 chars="v" unmodchars="v" repeat=0 keyCode=9
Press: native_key: 0x9 (v) glfw_key: 0x76 mods: none char_count: 1 deadKeyState: 0 repeat: 0 
	insertText: v replacementRange: (9223372036854775807, 0)
text: 0x76 glfw_key: v marked_text: ()
on_key_input: glfw key: 0x76 native_code: 0x9 action: PRESS mods: none text: 'v' state: 0 sent key as text to child: v
----------------- key up --------------------
NSEvent: type=KeyUp loc=(477.051,324.605) time=440990.9 flags=0x100 win=0x108724460 winNum=60958 ctxt=0x0 chars="v" unmodchars="v" repeat=0 keyCode=9
Release: native_key: 0x9 (v) glfw_key: 0x76 mods: none 
on_key_input: glfw key: 0x76 native_code: 0x9 action: RELEASE mods: none text: '' state: 0 ignoring as keyboard mode does not support encoding this event
```

**Comment by jackielii** [comment]
> That should be fixed by: [d63b852](https://github.com/kovidgoyal/kitty/commit/d63b852b90ce4818ae940c02c415b2cc0b759240)

Yes! I confirm it's fixed!
